airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -242,6 +267,7 @@ def list_keys(self, bucket_name=None, prefix='', delimiter='',
         return None
 
     @provide_bucket_name
+    @unify_bucket_name_and_key
```

Comment:
Do we need two wrappers? Do we use one without the other?

Reply:
Yes, because not every function that has a bucket has also a key as a parameter. I think it makes sense to split them.

Reply:
Does the order of decorators matter?

Reply:
yes. If you provide the bucket_name first from the connection, in case there is one, it won't extract it from the key.

Reply:
I think it's something worth a note somewhere

Reply:
Hmm. But where? I can't think of a good place for it.